src/instructlab/model/train.py

```diff
@@ -304,6 +305,28 @@ def train(
     Takes synthetic data generated locally with `ilab data generate` and the previous model and learns a new model using the MLX API.
     On success, writes newly learned model to {model_dir}/mlx_model, which is where `chatmlx` will look for a model.
     """
+
+    if not legacy and os.path.isdir(data_path):
+        # need to get msgs jsonl, this is a hack but will work for now
+        pattern = os.path.join(data_path, f'*{"messages"}*')
+
+        # Find all files matching the pattern
+        files = glob.glob(pattern)
+
+        # If no files are found, error
+        if not files:
+            ctx.fail(
+                "'--data-path' needs to be a valid jsonl when not using legacy training"
+            )
+
+        # Sort files by modification time (most recent first)
+        files.sort(key=lambda x: os.path.getmtime(x), reverse=True)
```

Comment:
Want to elevate this. I agree with the assessment here. The lambda is unnecessary.

Reply:
To provide some extra context here, the comparison function can just be passed as reference `os.path.getmtime` directly here via:

```suggestion
        files.sort(key=os.path.getmtime, reverse=True)
```